### PR TITLE
Easyxt Overhaul

### PR DESCRIPTION
**Pull request by lacan**
Methods are not contained in internal subclasses for ease of use.

### CODE REVIEW

**Review by romainGuiet — COMMENTED** [review]
minor changes required

File: src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java [hunk 1]
```diff
@@ -30,16 +30,16 @@
...
                     .setLowerThreshold(lowerThreshold)
                     .setName(surfaceName)
-                    .setColor(new Integer[]{color.getRed(),color.getGreen(),color.getBlue()})
+                    .setColor(new Integer[]{color.getRed(), color.getGreen(), color.getBlue()})
                     .build()
                     .detect();
 
             // Adds the surface to the scene
-            EasyXT.getImaris().GetSurpassScene().AddChild(surf,0);
+            EasyXT.Scene.addToScene(surf);
```
Comment (romainGuiet) [inline src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java:42]: following previous comment  "EasyXT.Scene.addToScene(surf)" could be changed to EasyXT.Scene.add(surf)

**Review by romainGuiet — COMMENTED** [review]
besides the discussion about how to name some functions, looks good to me!

File: src/main/java/ch/epfl/biop/imaris/EasyXT.java [hunk 2]
```diff
@@ -236,10 +229,8 @@ public static ISpotsPrx getSpots(String name) throws Error {
      * @throws Error an Imaris Error Object
      */
     public static ISpotsPrx getSpots(int position) throws Error {
-        ItemQuery query = new ItemQuery.ItemQueryBuilder().setType("Spots").setPosition(position).build();
-        List<IDataItemPrx> items = query.get();
-        if (items.size() > 0) return (ISpotsPrx) items.get(0);
-        return null;
+        ItemQuery query = new ItemQuery.ItemQueryBuilder().setType("Spots").build();
+        return (ISpotsPrx) query.get(0);
```
[the remark this answers is not in the document]
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/EasyXT.java]: removed position based search

File: src/main/java/ch/epfl/biop/imaris/ItemQuery.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
  * Copyright (c) 2020 Ecole Polytechnique Fédérale de Lausanne. All rights reserved.
  * Redistribution and use in source and binary forms, with or without modification,
  * are permitted provided that the following conditions are met:
- *
+ * <p>
```
[the remark this answers is not in the document]
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/ItemQuery.java:5]: hopefully done

File: src/main/java/ch/epfl/biop/imaris/command/GetSurfaceCommand.java [hunk 4]
```diff
@@ -22,9 +22,9 @@
     public void run() {
         // Gets an existing surface
         try {
-            ISurfacesPrx surfprx = EasyXT.getSurfaces( surfaceName );
+            ISurfacesPrx surfprx = EasyXT.Surfaces.getSurfaces(surfaceName);
             // Display surfaces
-            surface = EasyXT.getSurfacesMask( surfprx );
+            surface = EasyXT.Surfaces.getSurfacesMask(surfprx);
```
[the remark this answers is not in the document]
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/command/GetSurfaceCommand.java]: renamed

File: src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java [hunk 5]
```diff
@@ -30,16 +30,16 @@
     @Override
     public void run() {
         try {
-            ISurfacesPrx surf = SurfacesDetector.Channel(channelIndex)
+            ISurfacesPrx surf = EasyXT.Surfaces.create(channelIndex)
```
[the remark this answers is not in the document]
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java:32]: changed all names to reflect this

File: src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java (hunk 1 above)
Comment (lacan) [reply src/main/java/ch/epfl/biop/imaris/command/MakeSurfaceCommand.java]: changed to `Scene.putItem()`

**Comment by lacan** [comment]
Updated based on comments and much more

**Comment by lacan** [comment]
@romainGuiet think we are good to merge?

**Pull request merged**